.eslintrc.js

```diff
@@ -1,32 +1,110 @@
 /* eslint-env node */
-/** @type { import("eslint").Linter.Config } */
+/** @type {import('eslint').Linter.Config} */
 module.exports = {
+  parser: "@typescript-eslint/parser",
   env: {
     browser: true,
-    es6: true,
+    commonjs: true,
+    es2017: true,
   },
   extends: [
     "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:wc/recommended",
     "plugin:lit/recommended",
     "prettier",
-    "plugin:@typescript-eslint/recommended",
   ],
   plugins: ["@typescript-eslint", "lit"],
-  parser: "@typescript-eslint/parser",
   parserOptions: {
-    ecmaVersion: 2018,
-    sourceType: "module",
+    project: ["./tsconfig.eslint.json"],
+    tsconfigRootDir: __dirname,
   },
+  root: true,
   rules: {
+    /* start stylistic rules */
```

Comment:
Are the additional rules here mostly migration purposes?

Reply:
Yeah, that and keeping code quality high moving forward as well — most of these rules have auto-fixable suggestions, so I think this should be a good improvement in that regard without much added friction